app/views/devise/shared/_links.erb

```diff
@@ -1,12 +1,19 @@
+
 <%- if devise_mapping.omniauthable? and request.base_url != 'http://0.0.0.0:3000' %>
   <div class="small_margin_bottom">
     <%- resource_class.omniauth_providers.each do |provider| %>
-      <% if provider.to_s == "google_oauth2" && (current_page?(new_user_session_path) || current_page?(new_user_password_path)) %>
-        <%= button_to t('devise.shared.sign_in_google'), omniauth_authorize_path(resource_name, provider) %>
+      <% if (current_page?(new_user_session_path) || current_page?(new_user_password_path))%>
+         <% if provider.to_s == "google_oauth2" %>
+          <%= button_to t('devise.shared.sign_in_google'), user_google_oauth2_omniauth_authorize_path %>
+        <% elsif provider.to_s == "facebook"  %>
+          <%= button_to t('devise.shared.sign_in_facebook'), user_facebook_omniauth_authorize_path %>
+        <% end %>
       <% elsif provider.to_s == "google_oauth2" %>
-        <%= button_to t('devise.shared.sign_up_google'), omniauth_authorize_path(resource_name, provider) %>
-      <% end -%>
-    <% end -%>
+          <%= button_to t('devise.shared.sign_up_google'), user_google_oauth2_omniauth_authorize_path %>
+      <% else provider.to_s == "facebook" %>
```

Comment:
Should this be elsif too?

Reply:
Good point, you're right. Thanks, @julianguyen. I'll also create a new Issue for the env vars script, just in case someone can work on it before me.